spec/lib/diaspora_federation/validators/embed_validator_spec.rb

```diff
@@ -0,0 +1,40 @@
+module DiasporaFederation
+  describe Validators::EmbedValidator do
+    let(:entity) { :embed_entity }
+    it_behaves_like "a common validator"
+
+    describe "#url" do
+      it_behaves_like "a property with a value validation/restriction" do
+        let(:property) { :url }
+        let(:wrong_values) { %w[https://asdf$%.com example.com] }
+        let(:correct_values) { [nil, "https://example.org", "https://example.org/index.html"] }
+      end
+    end
+
+    describe "#title" do
+      it_behaves_like "a length validator" do
+        let(:property) { :title }
+        let(:length) { 255 }
+      end
+    end
+
+    describe "#description" do
+      it_behaves_like "a length validator" do
+        let(:property) { :description }
+        let(:length) { 65_535 }
+      end
+    end
+
+    describe "#image" do
+      it_behaves_like "a property with a value validation/restriction" do
+        let(:property) { :image }
+        let(:wrong_values) { %w[https://asdf$%.com example.com] }
+        let(:correct_values) { [nil] }
+      end
+
+      it_behaves_like "a url path validator" do
```

Comment:
Nit: I noticed that the ["url path validator"](https://github.com/diaspora/diaspora_federation/blob/32a49cc549e47b1e9239dd1d474f8de6c902a8a7/spec/support/shared_validator_specs.rb#L279) is missing example for the valid case with URL with a path:
```ruby
  it "validates url with a path" do
    validator = described_class.new(entity_stub(entity, property => "https://example.com/some/path"))

    expect(validator).to be_valid
    expect(validator.errors).to be_empty
  end
```

Might be worth adding?